python/tskit/trees.py

```diff
@@ -5952,6 +5954,26 @@ def sites_position(self):
         """
         return self._sites_position
 
+    @property
+    def sites_ancestral_state(self):
+        """
+        The ``ancestral_state`` column in the
+        :ref:`sec_site_table_definition` as a numpy array (dtype=StringDtype).
+        """
+        if self._sites_ancestral_state is None:
```

Comment:
This feels overly complicated. Why not:
```
if not not _tskit.HAS_NUMPY_2:
    raise RuntimeError(...)
if self._sites_ancestral_state is None:
     self._sites_ancestral_state = ...
return self._sites_ancestral_state
```
It doesn't look like there's any additional information in the ``raise`` path, and you may as well just let the error happen. Is the performance cost of looking up `` _tskit.HAS_NUMPY_2`` a concern or something?

Reply:
Let's switch so